src/core/utils/multicall.ts

```diff
@@ -79,47 +81,89 @@ export class MulticallQueue {
   }
 
   private async _flush() {
-    const callsToProcess = [...this.pendingCalls]
-    this.pendingCalls = []
+    const currentPendingCalls: PendingCallsMap = this.pendingCalls
+    this.pendingCalls = {}
 
-    const batchSize = callsToProcess.length
-    logger.debug({ batchSize }, 'Sending multicall batch ')
+    for (const [blockTag, callsToProcess] of Object.entries(
+      currentPendingCalls,
+    )) {
+      const batchSize = callsToProcess.length
+      logger.debug({ batchSize }, 'Sending multicall batch ')
 
-    const results = await this.multicallContract.aggregate3.staticCall(
-      callsToProcess.map(({ callParams }) => callParams),
-    )
-
-    const resultLength = results.length
-    const callsToProcessLength = callsToProcess.length
+      const results = await this.multicallContract.aggregate3.staticCall(
+        callsToProcess.map(({ callParams }) => callParams),
+        {
+          blockTag: blockTag == LATEST ? undefined : blockTag,
+        },
+      )
 
-    if (resultLength !== callsToProcessLength) {
-      // reject all to be on safe side
-      callsToProcess.forEach(({ reject }) => {
-        reject('Multicall batch failed')
-      })
+      const resultLength = results.length
+
+      if (resultLength !== batchSize) {
+        // reject all to be on safe side
+        callsToProcess.forEach(({ reject }) => {
+          reject({ message: 'Multicall batch failed' })
```

Comment:
JS or TS don't enforce it (thus why the param for reject is `reason?: any`, but it is good practice to always throw/reject errors (which will indeed have `message: string` as per their interface,
```suggestion
          reject(new Error('Multicall batch failed'))
```